packages/@react-types/shared/src/dom.d.ts

```diff
@@ -160,19 +157,3 @@ export interface TextInputDOMProps extends DOMProps {
    */
   onInput?: FormEventHandler<HTMLInputElement>
 }
-
-/**
- * A map of HTML element names and their interface types.
- * For example `'a'` -> `HTMLAnchorElement`.
- */
-export type IntrinsicHTMLElements = {
-  [K in keyof IntrinsicHTMLAttributes]: IntrinsicHTMLAttributes[K] extends HTMLAttributes<infer T> ? T : never
-};
-
-/**
- * A map of HTML element names and their attribute interface types.
- * For example `'a'` -> `AnchorHTMLAttributes<HTMLAnchorElement>`.
- */
-export type IntrinsicHTMLAttributes = {
-  [K in keyof ReactDOM]: ReactDOM[K] extends DOMFactory<infer T, any> ? T : never
-};
```

Comment:
Just a heads-up, the plan was to use these in the `useButton` refactoring, similar to what we did for `useTextField`; i.e. making `useButton` generic instead of overloaded.

Reply:
Ok let's talk about that then. I didn't want to get locked into exposing this as API forever. e.g. maybe it's possible to get these types from somewhere, e.g. react itself?